server/backend/mlx_runner.py

```
@@ -769,7 +975,15 @@ def generate_batch(
         )
 
         # Format reasoning models output
+        # TODO: this is where we are supposed to use harmony
+        print(f"Model response \n{response}")
+
+        # encoding.parse_messages_from_completion_tokens()
+        print(
+            f"\n harmonied response\n{encoding.parse_messages_from_completion_tokens(all_tokens, Role.ASSISTANT)}"
```

Ruff (0.15.2)

[error] Undefined name `encoding`

(F821)